src/python/gem5/simulate/simulator.py

```diff
@@ -91,6 +97,32 @@ class Simulator:
         "compatible with the gem5 standard library.",
     }
 
+    def get_default_exit_handler_id_map(cls) -> Dict[int, Type[ExitEvent]]:
+        default_map = {
+            # The default exit handlers for the older "generated-based" exit
+            # events.
+            0: ClassicGeneratorExitHandler,
+            # The default exit handler for when the the kernel in a FS
+            # simulation has completed booting.
+            1: KernelBootedExitHandler,
+            # The default exit handler for when the `after_boot.sh` script has
+            # completed execution in FS simulations which utilize this.
+            2: AfterBootExitHandler,
+            # The default exit handler triggered when the post-boot script has
+            # completed execution in FS simulation. Note: it is common for this
+            # to be the the defacto end of the simulation as the post-boot
+            # script is used to execute the user's workload.
+            3: AfterBootScriptExitHandler,
+            # The default exit handler for scheduled exit event, such as those
+            # scheduled by the user via `scheduleTickExitAbsolute` or
+            # `scheduleTickExitFromCurrent`.
+            4: ScheduledExitEventHandler,
+        }
+        assert all(
+            i >= 0 for i in default_map.keys()
+        ), "Exit handler mapped to ID <0 in default map."
+        return default_map
```

Comment:
I still think this can automatically be generated (and extended) without having to modify the stdlib code (and rebuild gem5). I can mock this up on Thursday.

Reply:
This can be extended and modified without rebuilding gem5. This is just the default allocation but it can be extended/modified via the configuration script as needed..